sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5445,19 +5447,43 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
     ContextsLock.lock();
     Queue->CaptureIndirectAccesses();
     // Add the command to the command list, which implies submission.
-    ZE_CALL(zeCommandListAppendLaunchKernel,
-            (CommandList->first, Kernel->ZeKernel, &ZeThreadGroupDimensions,
-             ZeEvent, (*Event)->WaitList.Length,
-             (*Event)->WaitList.ZeEventList));
+    ZeResult = ZE_CALL_NOCHECK(
+        zeCommandListAppendLaunchKernel,
+        (CommandList->first, Kernel->ZeKernel, &ZeThreadGroupDimensions,
+         ZeEvent, (*Event)->WaitList.Length, (*Event)->WaitList.ZeEventList));
   } else {
     // Add the command to the command list for later submission.
     // No lock is needed here, unlike the immediate commandlist case above,
     // because the kernels are not actually submitted yet. Kernels will be
     // submitted only when the comamndlist is closed. Then, a lock is held.
-    ZE_CALL(zeCommandListAppendLaunchKernel,
-            (CommandList->first, Kernel->ZeKernel, &ZeThreadGroupDimensions,
-             ZeEvent, (*Event)->WaitList.Length,
-             (*Event)->WaitList.ZeEventList));
+    ZeResult = ZE_CALL_NOCHECK(
+        zeCommandListAppendLaunchKernel,
+        (CommandList->first, Kernel->ZeKernel, &ZeThreadGroupDimensions,
+         ZeEvent, (*Event)->WaitList.Length, (*Event)->WaitList.ZeEventList));
+  }
+
+  if (ZeResult != ZE_RESULT_SUCCESS) {
```

Comment:
Please add a TODO comment that we need to re-do this in a robust way.

Reply:
We decided to abandon this patch and rely on the [optional kernel features](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:optional-kernel-features) implementation.
Sorry for the mess with closing/opening this PR.